lib/browser/rpc-server.js

```diff
@@ -440,20 +442,47 @@ ipcMain.on('ELECTRON_BROWSER_WINDOW_CLOSE', function (event) {
   event.returnValue = null
 })
 
-ipcMain.on('ELECTRON_BROWSER_SANDBOX_LOAD', function (event, preloadPath) {
+ipcMain.on('ELECTRON_BROWSER_SANDBOX_LOAD', function (event) {
+  let preloadPath = null
   let preloadSrc = null
   let preloadError = null
+  try {
+    preloadPath = getPreloadPath(event.sender)
+  } catch (err) {
+    preloadError = err || new Error('Failed to get preload script path')
+  }
   if (preloadPath) {
     try {
       preloadSrc = fs.readFileSync(preloadPath).toString()
     } catch (err) {
-      preloadError = {stack: err ? err.stack : (new Error(`Failed to load "${preloadPath}"`)).stack}
+      preloadError = err || new Error(`Failed to load "${preloadPath}"`)
     }
   }
   event.returnValue = {
     preloadSrc: preloadSrc,
-    preloadError: preloadError,
+    preloadError: preloadError ? {stack: preloadError.stack} : null,
     platform: process.platform,
     env: process.env
   }
 })
+
+const getPreloadPath = function (webContents) {
```

Comment:
equivalent of the code in https://github.com/electron/electron/blob/master/atom/browser/web_contents_preferences.cc

```
  // The preload script.
  base::FilePath::StringType preload;
  if (GetAsString(&preference_, options::kPreloadScript, &preload)) {
    if (base::FilePath(preload).IsAbsolute())
      command_line->AppendSwitchNative(switches::kPreloadScript, preload);
    else
      LOG(ERROR) << "preload script must have absolute path.";
  } else if (GetAsString(&preference_, options::kPreloadURL, &preload)) {
    // Translate to file path if there is "preload-url" option.
    base::FilePath preload_path;
    if (net::FileURLToFilePath(GURL(preload), &preload_path))
      command_line->AppendSwitchPath(switches::kPreloadScript, preload_path);
    else
      LOG(ERROR) << "preload url must be file:// protocol.";
  }
```